cli/global_state.rs

```diff
@@ -113,54 +120,91 @@ impl GlobalState {
   ) -> Result<(), AnyError> {
     let module_specifier = module_specifier.clone();
 
-    let mut module_graph_loader = ModuleGraphLoader::new(
-      self.file_fetcher.clone(),
-      maybe_import_map,
-      permissions.clone(),
-      is_dyn_import,
-      false,
-    );
-    module_graph_loader
-      .add_to_graph(&module_specifier, maybe_referrer)
-      .await?;
-    let module_graph = module_graph_loader.get_graph();
-
-    let out = self
-      .file_fetcher
-      .fetch_cached_source_file(&module_specifier, permissions.clone())
-      .expect("Source file not found");
-
-    let module_graph_files = module_graph.values().collect::<Vec<_>>();
-    // Check integrity of every file in module graph
-    if let Some(ref lockfile) = self.lockfile {
-      let mut g = lockfile.lock().unwrap();
+    if self.flags.no_check {
+      debug!("Transpiling root: {}", module_specifier);
+      let handler =
+        Rc::new(RefCell::new(FetchHandler::new(&self.flags, &permissions)?));
+      let mut builder = GraphBuilder::new(handler, maybe_import_map);
+      builder.insert(&module_specifier).await?;
+      let mut graph = builder.get_graph();
+
```

Comment:
Looks like integrity check using lock file is not done in this path

Reply:
Ah, good point.  Need to do that as part of the `SpecifierHandler`.

Reply:
Actually when I looked at it, the best point to do it was when moving the graph out of the builder, as that _finalizes_ the graph.  I've added that now.